include/text.php

```diff
@@ -1196,6 +1197,11 @@ function put_item_in_cache(&$item, $update = false)
 		$item["rendered-html"] = prepare_text($item["body"]);
 		$item["rendered-hash"] = hash("md5", $item["body"]);
 
+		// Force an update if the generated values differ from the existing ones
+		if (($rendered_hash != $item["rendered-hash"]) || ($rendered_html != $item["rendered-html"])) {
```

Comment:
The hash is used to prevent a full-text comparison, so either the `rendered-html` check is superfluous, either the hashing is superfluous.

Reply:
I had a rendering issue where the BBCode stayed the same but the HTML differed. When coding with the BBCode parser this is important.

Additionally the comparism with the HTML is the second one of an "or" statement. So the second comparism will only be executed if the first one doesn't differ.